src/de/serde.rs

```diff
@@ -562,11 +562,58 @@ where
 /// Serde Deserializer
 pub struct Deserializer {
     value: Option<Bson>,
+    options: DeserializerOptions,
+}
+
+/// Options used to configure a [`Deserializer`]. These can also be passed into
+/// [`crate::from_bson_with_options`] and [`crate::from_document_with_options`].
+#[derive(Debug, Clone, Default)]
+#[non_exhaustive]
+pub struct DeserializerOptions {
+    /// Whether the [`Deserializer`] should present itself as human readable or not.
+    /// The default is true.
+    pub human_readable: Option<bool>,
+}
+
+impl DeserializerOptions {
+    /// Create a builder struct used to construct a [`DeserializerOptions`].
+    pub fn builder() -> DeserializerOptionsBuilder {
+        DeserializerOptionsBuilder {
+            options: Default::default(),
+        }
+    }
+}
+
+/// Builder used to construct a [`DeserializerOptions`].
+pub struct DeserializerOptionsBuilder {
```

Comment:
Since it was just a single option, I opted to create the builder manually rather than bringing in `typed-builder`.